examples/watch_resources/watch_test.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch_resources
+
+import (
+	"context"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/k8s/watcher"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestWatchForResources(t *testing.T) {
+	watchFeature := features.New("test watcher").WithLabel("env", "dev").
```

Comment:
Because method `Start` now does the event processing in a goroutine, it is important to ensure that the callback methods are firing. So, I am proposing updating the test as follows


```go
func TestWatchForResourcesWithStop(t *testing.T) {
    addWait := make(chan struct{})
    onAdd := func (obj interface{}) {
	dep := obj.(*appsv1.Deployment)
	depName := dep.GetName()
	if depName == "watch-dep"  {
		 addWait <- struct{}{}
	}
    }
    ...
    Assess("create watch...", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
        ...
	if err := client.Resources().Create(ctx, deployment); err != nil {
	    t.Fatal(err)
	}
        // After creation, wait for signal
        select {
            case <-time.After(3 *time.Second):
            t.Error("Add callback not called)
            case <-addWait:
            close(addWait)
        }
        ...
        return context.WithValue(ctx, "test-dep", deployment)
    })
}
```

You can also do something similar to watch for deleted messaging firing.